### PR TITLE
Update RTP SSRC mapping table for MID found  in RTCP SDES

### PR DESCRIPTION
**Pull request by fluffy**
Fixes https://github.com/rtcweb-wg/jsep/issues/342 https://github.com/rtcweb-wg/jsep/issues/348

### CODE REVIEW

File: draft-ietf-rtcweb-jsep.xml [hunk 1]
```diff
@@ -3640,6 +3640,20 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
         </list>
       </t>
 
+      <t>For each RTCP packet received, the following steps MUST be
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: We probably need a more general treatment of how to route RTCP. That is, how do we route normal RTCP packets in a bundle scenario?

File: draft-ietf-rtcweb-jsep.xml [hunk 2]
```diff
@@ -3646,11 +3650,28 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
       <t>
         <list>
 
-	  <t>If the RTCP packet has a SDES with a MID, and that MID is
-	  in the table mapping MID to RtpReceiver, update the SSRC
-	  mapping table to include an entry mapping the packet's SSRC
-	  to the RtpReceiver.</t>
-
+	  <t>For each SDES item in the RTCP packet, if the RTCP packet
+	  has a SDES with a MID, and that MID is in the table mapping
```
Comment (taylor-b) [inline draft-ietf-rtcweb-jsep.xml]: "an SDES"

File: draft-ietf-rtcweb-jsep.xml [hunk 3]
```diff
@@ -3646,11 +3650,28 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+	  MID to a RtpReceiver, update the incoming SSRC mapping table
```
Comment (taylor-b) [inline draft-ietf-rtcweb-jsep.xml]: "an RtpReceiver", or just "MID to RtpReceiver"

File: draft-ietf-rtcweb-jsep.xml [hunk 4]
```diff
@@ -3640,6 +3644,37 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+
+	  <t>For each SDES item in the RTCP packet, if the RTCP packet
+	  has a SDES with a MID, and that MID is in the table mapping
+	  MID to a RtpReceiver, update the incoming SSRC mapping table
+	  to include an entry mapping the SSRC for that SDES item to
+	  the RtpReceiver associated with the MID.</t>
+
+	  <t> For each Receiver Report in the RTCP packet, if the SSRC
+	  for the Receiver Report is found in the outgoing SSRC table,
+	  forward a copy of the RTCP packet to the RtpReceiver found
```
Comment (taylor-b) [inline draft-ietf-rtcweb-jsep.xml]: nit: Elsewhere it looks like we say "deliver", not "forward"

Also, deliver to the RtpSender here, not receiver.

File: draft-ietf-rtcweb-jsep.xml [hunk 5]
```diff
@@ -3670,7 +3705,9 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
       processing of the RTP packets is done at the RtpReceiver level.
       This includes using
       <xref target="I-D.ietf-mmusic-rid" /> to determine which RTP
-      streams depend on or repair other RTP streams.</t>
+      streams depend on or repair other RTP streams. If the RTP packet
+      did not have PT that matched any codec for this RtpReceiver, the
+      packer would be dropped. </t>
```
Comment (taylor-b) [inline draft-ietf-rtcweb-jsep.xml]: packet

File: draft-ietf-rtcweb-jsep.xml [hunk 6]
```diff
@@ -3646,11 +3650,28 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-
+	  <t>For each SDES item in the RTCP packet, if the RTCP packet
+	  has a SDES with a MID, and that MID is in the table mapping
+	  MID to a RtpReceiver, update the incoming SSRC mapping table
+	  to include an entry mapping the SSRC for that SDES item to
+	  the RtpReceiver associated with the MID.</t>
+
+	  <t> For each Receiver Report in the RTCP packet, if the SSRC
+	  for the Receiver Report is found in the outgoing SSRC table,
+	  forward a copy of the RTCP packet to the RtpReceiver found
```
Comment (taylor-b) [inline draft-ietf-rtcweb-jsep.xml]: nit: Elsewhere it looks like we say "deliver", not "forward"

Also, deliver to the RtpSender here, not receiver?

File: draft-ietf-rtcweb-jsep.xml [hunk 7]
```diff
@@ -3623,12 +3623,16 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
           <t>Construct a table mapping MID to RtpReceiver for each
           RtpReceiver configured to receive from this transport.</t>
 
-          <t>Construct a table mapping SSRC to RtpReceiver for each
+          <t>Construct a table mapping incoming SSRC to RtpReceiver for each
           RtpReceiver configured to receive from this transport and for
           each SSRC that RtpReceiver is configured to receive. Some of
           the SSRCs may be presesnt in the m= section corresponding to
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: NIT: present

File: draft-ietf-rtcweb-jsep.xml [hunk 8]
```diff
@@ -3623,12 +3623,16 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
           that RtpReceiver in the remote description.</t>
 
+	  <t>Construct a table mapping outgoing SSRC to RtpSender for each
+          RtpSender configured to transmit from this transport and for
+          each SSRC that RtpSender is configured to use when sending. </t>
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: NIT: extra space before </t>

File: draft-ietf-rtcweb-jsep.xml [hunk 9]
```diff
@@ -3640,6 +3644,37 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
         </list>
       </t>
 
+      <t>For each RTCP packet received, the following steps MUST be
+      followed:</t>
+      
+      <t>
+        <list>
+
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: NIT: Whitespace

File: draft-ietf-rtcweb-jsep.xml [hunk 10]
```diff
@@ -3640,6 +3644,37 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+	  <t>For each SDES item in the RTCP packet, if the RTCP packet
+	  has a SDES with a MID, and that MID is in the table mapping
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: Do you mean to say "For each SDES item in the RTCP packet that contains a MID"

File: draft-ietf-rtcweb-jsep.xml [hunk 11]
```diff
@@ -3640,6 +3644,37 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+	  MID to a RtpReceiver, update the incoming SSRC mapping table
+	  to include an entry mapping the SSRC for that SDES item to
+	  the RtpReceiver associated with the MID.</t>
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: perhaps
"update the incoming SSRC mapping table to include an entry mapping that SSRC to the RtpReceiver associated with the MID"

File: draft-ietf-rtcweb-jsep.xml [hunk 12]
```diff
@@ -3640,6 +3644,37 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+
+	  <t> For each Receiver Report in the RTCP packet, if the SSRC
+	  for the Receiver Report is found in the outgoing SSRC table,
+	  deliver a copy of the RTCP packet to the RtpSender found
+	  in the outgoing SSRC table.  </t>
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: Nit: space before </t>

File: draft-ietf-rtcweb-jsep.xml [hunk 13]
```diff
@@ -3640,6 +3644,37 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+
+	  <t> For each Sender Report in the RTCP packet, if the SSRC
+	  for the Sender Report is found in the incoming SSRC table,
+	  deliver a copy of the RTCP packet to the RtpReceiver found
+	  in the incoming SSRC table.  </t>
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: NIT: whitespace again

File: draft-ietf-rtcweb-jsep.xml [hunk 14]
```diff
@@ -3670,7 +3705,9 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
       processing of the RTP packets is done at the RtpReceiver level.
       This includes using
       <xref target="I-D.ietf-mmusic-rid" /> to determine which RTP
-      streams depend on or repair other RTP streams.</t>
+      streams depend on or repair other RTP streams. If the RTP packet
+      did not have PT that matched any codec for this RtpReceiver, the
+      packet would be dropped. </t>
```
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: I think you mean to say
"If the payload type in the RTP packet does not match any codec for this RtpReceiver, the packet will be dropped"

File: draft-ietf-rtcweb-jsep.xml [hunk 15]
```diff
@@ -3646,35 +3646,32 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-	  in the incoming SSRC table.  </t>
-
-	  <t> For any BYE or APP RTCP packets, for each SSRC and CSRC
-	  in the BYE or APP packet, if the value of the SSRC or CSRC
-	  is found in the incoming SSRC table, deliver a copy of the
-	  RTCP packet to the RtpReceiver in the incoming SSRC
-	  table. </t>
-	  
-	</list>
+          <t>For each SDES item in the RTCP packet which contains a
```
Comment (taylor-b) [inline draft-ietf-rtcweb-jsep.xml]: nit: "that" instead of "which"

File: draft-ietf-rtcweb-jsep.xml [hunk 16]
```diff
@@ -3646,35 +3646,32 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+          MID that is in the table mapping MID to a RtpReceiver, update the
+          incoming SSRC mapping table to include an entry the SSRC for that SDED
```
Comment (fluffy) [inline draft-ietf-rtcweb-jsep.xml]: SDED -> SDES

Comment (taylor-b) [inline draft-ietf-rtcweb-jsep.xml]: SDES

File: draft-ietf-rtcweb-jsep.xml [hunk 17]
```diff
@@ -3646,35 +3646,32 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+          item to the RtpReceiver associated with the MID.</t>
+
+          <t> For each Receiver Report in the RTCP packet, if the SSRC
```
Comment (taylor-b) [inline draft-ietf-rtcweb-jsep.xml]: nit: don't need spaces after "\<t\>"

File: draft-ietf-rtcweb-jsep.xml [hunk 18]
```diff
@@ -3705,9 +3702,9 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
       processing of the RTP packets is done at the RtpReceiver level.
       This includes using
       <xref target="I-D.ietf-mmusic-rid" /> to determine which RTP
-      streams depend on or repair other RTP streams. If the RTP packet
-      did not have PT that matched any codec for this RtpReceiver, the
-      packet would be dropped. </t>
+      streams depend on or repair other RTP streams. If the RTP packet's
+      PT does not match any codec in use at the RtpReceiver, the
```
Comment (taylor-b) [inline draft-ietf-rtcweb-jsep.xml]: at the -> by the?

File: draft-ietf-rtcweb-jsep.xml [hunk 19]
```diff
@@ -3623,20 +3623,55 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
           <t>Construct a table mapping MID to RtpReceiver for each
           RtpReceiver configured to receive from this transport.</t>
 
-          <t>Construct a table mapping SSRC to RtpReceiver for each
-          RtpReceiver configured to receive from this transport and for
-          each SSRC that RtpReceiver is configured to receive. Some of
-          the SSRCs may be presesnt in the m= section corresponding to
-          that RtpReceiver in the remote description.</t>
+          <t>Construct a table mapping incoming SSRC to RtpReceiver
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: suggest adjusting this section title to mention RTP/RTCP packets

File: draft-ietf-rtcweb-jsep.xml [hunk 20]
```diff
@@ -3623,20 +3623,55 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-          local description. If any payload type could map to more than
-          one RtpReceiver, map to the RtpReceiver whose m= section
-          appears earliest in the local description.</t>
+          local description. If any payload type could map to more
+          than one RtpReceiver, map to the RtpReceiver whose m=
+          section appears earliest in the local description.</t>
+        </list>
+      </t>
+
+      <t>For each RTCP packet received, the following steps MUST be
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Suggest having this section come after the RTP section

File: draft-ietf-rtcweb-jsep.xml [hunk 21]
```diff
@@ -3623,20 +3623,55 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+      followed:</t>
+
+      <t>
+        <list>
+          <t>For each SDES item in the RTCP packet that contains a
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: This is a special case and as such I think this para should come last in this list.

File: draft-ietf-rtcweb-jsep.xml [hunk 22]
```diff
@@ -3623,20 +3623,55 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+          MID that is in the table mapping MID to RtpReceiver, update
+          the incoming SSRC mapping table to include an entry the SSRC
+          for that SDES item to the RtpReceiver associated with the
+          MID.</t>
+
+          <t>For each Receiver Report in the RTCP packet, if the SSRC
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: There can be multiple SSRCs in a RTCP packet, so I suggest spelling out "sender SSRC" (here and below)

File: draft-ietf-rtcweb-jsep.xml [hunk 23]
```diff
@@ -3623,20 +3623,55 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+          for the Receiver Report is found in the outgoing SSRC table,
+          deliver a copy of the RTCP packet to the RtpSender found in
+          the outgoing SSRC table.</t>
+
+          <t>For each Sender Report in the RTCP packet, if the SSRC
+          for the Sender Report is found in the incoming SSRC table,
+          deliver a copy of the RTCP packet to the RtpReceiver found
+          in the incoming SSRC table.</t>
+
+          <t>For any BYE or APP RTCP packets, for each SSRC and CSRC
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Need to consider PSFB and RTPFB as well (when sent as reduced-size).

File: draft-ietf-rtcweb-jsep.xml [hunk 24]
```diff
@@ -3623,20 +3623,55 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+          the incoming SSRC mapping table to include an entry the SSRC
+          for that SDES item to the RtpReceiver associated with the
+          MID.</t>
+
+          <t>For each Receiver Report in the RTCP packet, if the SSRC
+          for the Receiver Report is found in the outgoing SSRC table,
+          deliver a copy of the RTCP packet to the RtpSender found in
+          the outgoing SSRC table.</t>
+
+          <t>For each Sender Report in the RTCP packet, if the SSRC
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: I think this also needs to go to the RtpSender as well, since a SR contains all RR information as well.

**Comment by ekr** [comment]
LGTM

**Comment by juberti** [comment]
I made a bunch of edits to cover how the different types of RTCP packets should be handled, including compound RTCP packets. I also tweaked the RTP text slightly to be more consistent with the RTCP text.
@deadbeef, @fluffy, @ekr PTAL.

**Comment by juberti** [comment]
@taylor-b, can you take a look?

**Comment by fluffy** [comment]
LGTM

**Pull request merged**